src/main/java/org/eclipse/uprotocol/uuid/factory/UUIDUtils.java

```diff
@@ -21,163 +21,191 @@
+        return uuid == null ? Optional.empty() : Version.getVersion(uuid.version());
     }
 
     /**
-     * Verify uuid is either v6 or v8
-     * @return true if is UUID version 6 or 8
+     * Fetch the Variant from the passed UUID
+     * @param uuid The UUID to fetch the variant from
+     * @return Empty if uuid is null, otherwise the variant
      */
-    public static boolean isUuid(UUID uuid) { return isUProtocol(uuid) || isUuidv6(uuid); }
-
+    public static Optional<Integer> getVariant(UUID uuid) {
+        return uuid == null ? Optional.empty() : Optional.of(uuid.variant());
+    }
+    
     /**
      * Verify if version is a formal UUIDv8 uProtocol ID
      * @return true if is a UUIDv8 for uProtocol
      */
     public static boolean isUProtocol(UUID uuid) {
-        return getVersion(uuid) == Version.VERSION_UPROTOCOL;
+        final Optional<Version> version = getVersion(uuid);
+        return uuid == null || version.isEmpty() ? false : version.get() == Version.VERSION_UPROTOCOL;
     }
 
     /**
      * Verify if version is This one
      * @return true if is UUID version 6
      */
     public static boolean isUuidv6(UUID uuid) {
-        return getVersion(uuid) == Version.VERSION_TIME_ORDERED;
+        final Optional<Version> version = getVersion(uuid);
+        final Optional<Integer> variant = getVariant(uuid);
+        return uuid != null && !version.isEmpty() ? version.get() == Version.VERSION_TIME_ORDERED && 
+            variant.get() == UuidVariant.VARIANT_RFC_4122.getValue() : false;
     }
 
     /**
-     *  Return the number of miliseconds since unix epoch
+     * Verify uuid is either v6 or v8
+     * @return true if is UUID version 6 or 8
+     */
+    public static boolean isUuid(UUID uuid) { return isUProtocol(uuid) || isUuidv6(uuid); }
+
+    /**
+     * Return the number of milliseconds since unix epoch from a passed UUID
      * @param uuid passed uuid to fetch the time
-     * @return number of miliseconds since unix epoch
+     * @return number of milliseconds since unix epoch or empty if uuid is null
      */
-    public static long getTime(UUID uuid) {
-        final long time;
-        switch (getVersion(uuid)) {
+    public static Optional<Long> getTime(UUID uuid) {
+        Long time = null;
+        Optional<Version> version = getVersion(uuid);
+        if (uuid == null || version.isEmpty()) {
+            return Optional.empty();
+        }
+
+        switch (version.get()) {
             case VERSION_UPROTOCOL:
                 time = uuid.getMostSignificantBits() >> 16;
                 break;
             case VERSION_TIME_ORDERED:
-                time = UuidTime.toUnixTimestamp(UuidUtil.getTimestamp(uuid));
+                //convert Ticks to Millis
+                try {
+                    time =  UuidTime.toUnixTimestamp(UuidUtil.getTimestamp(uuid)) / UuidTime.TICKS_PER_MILLI;
+                } catch (IllegalArgumentException e) {
```

Comment:
exceptions being ignored  - need to at least log.
Rust implementations might want to use the Result "monad" instead of optional

Reply:
Per our conversation we will leave it as is for now